### PR TITLE
Add structs.

### PR DESCRIPTION
**Pull request by NJichev**
Generate constant atoms for the __struct__ field otherwise it gets passed as an atom.

### CODE REVIEW

File: test/stream_data/types_list.ex [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
+  defmodule SomeStruct do
+    defstruct [:key]
+  end
+
   @type basic_any() :: any()
   @type basic_atom :: atom()
   @type basic_none() :: none()
   @type basic_reference :: reference()
   @type basic_map :: map()
+  @type basic_struct :: struct()
```
Comment (whatyouhide) [inline test/stream_data/types_list.ex]: Where are we handling this? That is, what's the format this is represented with?

File: test/stream_data/types_test.exs [hunk 2]
```diff
@@ -67,6 +67,15 @@ defmodule StreamData.TypesTest do
       end
     end
 
+    test "struct" do
+      data = generate_data(:basic_struct)
+
+      check all x <- data, max_runs: 25 do
+        assert is_map(x)
+        assert %_{} = x
```
Comment (whatyouhide) [inline test/stream_data/types_test.exs]: We can remove the `is_map` test given this also checks for that.

File: test/stream_data/types_test.exs [hunk 3]
```diff
@@ -332,6 +341,25 @@ defmodule StreamData.TypesTest do
...
+
+    test "struct with all fields key type" do
+      data = generate_data(:literal_struct_all_fields_key_type)
+
+      check all x <- data, max_runs: 25 do
+        assert %StreamDataTest.TypesList.SomeStruct{key: value} = x
+        assert is_integer(value)
+      end
+    end
+
```
Comment (whatyouhide) [inline test/stream_data/types_test.exs]: Nitpick: kill this line :) ✂️

File: test/stream_data/types_list.ex (hunk 1 above)
Comment (NJichev) [reply test/stream_data/types_list.ex]: Structs are defined inside `elixir.erl` so it get's picked up as a remote type [here](https://github.com/NJichev/type-generators/blob/1d3a3a78f6f5061f81c3295e4b6586cf6af6d4b3/lib/stream_data_types.ex#L255).
And then it gets translated to a map with the following type "signature":

```
{:type, 21, :map,
 [
   {:type, 21, :map_field_exact,
    [{:atom, 21, :__struct__}, {:type, 21, :atom, []}]},
   {:type, 21, :map_field_assoc,
    [{:type, 21, :atom, []}, {:type, 21, :any, []}]}
 ]}
```
Which is already handled by the maps.

File: test/stream_data/types_test.exs (hunk 3 above)
Comment (NJichev) [reply test/stream_data/types_test.exs]: I should add the formatter to CI :sweat_smile:

**Pull request merged**